### ISSUE TITLE
When test, the printed log does not show the full text of the metrics, when named in Chinese

### ISSUE THREAD

**Issue by 123mutourener** [Lightning-AI/lightning#14837]
### First check

- [X] I'm sure this is a bug.
- [X] I've added a descriptive title to this bug.
- [X] I've provided clear instructions on how to reproduce the bug.
- [X] I've added a code sample.
- [X] I've provided any other important info that is required.

### Bug description

Hi,

I am dealing with a very messy segmentation problem, which requires me to log the per-category accuracy and iou at test time. I used self.log(category, metrics) to do this job at "on_test_epoch_end". However, the log does not show the full metrics text, and shorten rest of it to "..." instead, when the metric name is in Chinese. I know that in the file pytorch_lightning.loops.dataloader.evaluation_loop, by line 340, it setting the max_length based on the longest string. However, my longest Chinese metric is 22 words long, and I checked that "max_length" is 25, but the printed metrics still hides part of the name.

![Screenshot from 2022-09-22 14-43-52](https://user-images.githubusercontent.com/15895948/191676824-ad216c9b-3f22-479b-9798-6c547a60e49c.png)

 

### How to reproduce the bug

```python
def on_test_epoch_end(self):
    self.log("车道线-虚减速让行线_清晰_树荫遮挡_acc", 0.0121)
```

### Error messages and logs

```

# Error messages and logs here please

```

### Important info

```

#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): LightningModule
#- PyTorch Lightning Version (e.g., 1.5.0): 1.7.6
#- PyTorch Version (e.g., 1.10): 1.12.1
#- Python version (e.g., 3.9): 3.9
#- OS (e.g., Linux): Ubuntu 20.04 in English
#- CUDA/cuDNN version: 11.6
#- GPU models and configuration: 3060Ti
#- How you installed Lightning(`conda`, `pip`, source): pip
#- Running environment of LightningApp (e.g. local, cloud): local

```

**Comment by 123mutourener** [comment]
I have a solution, that is:
`matrics_encoded = [metric.encode("gbk") for metric in metrics_strs]
max_length = int(min(max(len(max(matrics_encoded, key=len)), len(max(headers, key=len)), 25), term_size / 2))
`

**Comment by awaelchli** [comment]
@123mutourener Thanks for giving this a try. I'm not familiar with the best practices around handling these differences in languages, but it looks like your solution improves on the existing implementation. The columns are unfortunately still broken, but at least it shows the full name of the metric:

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃              Test metric              ┃             DataLoader 0             ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│               test_loss               │          -2.555753707885742          │
│ 车道线-虚减速让行线_清晰_树荫遮挡_acc │         0.01209999993443489          │
└───────────────────────────────────────┴──────────────────────────────────────┘

```

Would you like to send a PR with your suggestion?

**Comment by 123mutourener** [comment]
Hi @awaelchli , yes I will submit a PR. In my case, the table looks fine. I run the code in Linux terminal.
